File: server/read_pdf.js
const fs = require('fs');
const pdf = require('pdf-parse');

const dataBuffer = fs.readFileSync('C:\\Users\\TUF\\Desktop\\Structure du rapport PFA.pdf');

pdf(dataBuffer).then(function(data) {
    console.log(data.text);
}).catch(function(error) {
    console.error('Error reading PDF:', error);
});

File: server/read_pdf.py
import PyPDF2
import sys

def read_pdf(file_path):
    try:
        with open(file_path, 'rb') as file:
            reader = PyPDF2.PdfReader(file)
            text = ''
            for page in reader.pages:
                text += page.extract_text() + '\n'
            with open("pdf_content.txt", "w", encoding="utf-8") as out_file:
                out_file.write(text)
    except Exception as e:
        print(f"Error reading PDF: {e}", file=sys.stderr)

if __name__ == "__main__":
    read_pdf(r"C:\Users\TUF\Desktop\Structure du rapport PFA.pdf")
